stix_shifter_modules/trellix_endpoint_security_hx/README.md

```diff
@@ -0,0 +1,697 @@
+                            "data_type": "REG_BINARY",
+                            "data": "............ ...................^....."
+                        }
+                    ]
+                },
+                "4": {
+                    "type": "x-oca-asset",
+                    "device_id": "device1",
+                    "hostname": "host1",
+                    "x_host_set": "host_set1"
+                },
+                "5": {
+                    "type": "file",
+                    "name": "explorer.exe"
+                }
+            },
+            "first_observed": "2024-05-29T15:54:50.795Z",
+            "last_observed": "2024-05-29T15:54:50.795Z",
+            "number_observed": 1
+        }
+    ],
+    "spec_version": "2.0"
+}
+```
+
+### Usage of host sets in Command Line
+
+- The host set name should be passed as comma\(,\) separated values in options parameter under connections 
+  while running the queries in command line.
+
+#### STIX Translate query
+
+translate trellix_endpoint_security_hx query "{}" "[domain-name:value LIKE 'download'] START t'2024-05-01T00:00:00.000Z' STOP t'2024-05-29T00:00:00.000Z'"
+"{\"host_sets\": \"host_set1,host_set2\"}"
+
+#### STIX Transmit Query
+
+transmit
+trellix_endpoint_security_hx
+"{\"host\":\"1.2.3.4\",\"port\":123,\"selfSignedCert\":\"cert\",\"options\":{\"host_sets\":\"host_set1,host_set2\"}}"
+"{\"auth\":{\"username\":\"xxx\",\"password\":  \"yyyy\"}}"
+query
+{input query}
+
+#### STIX Transmit Status
+
+transmit
+trellix_endpoint_security_hx
+"{\"host\":\"1.2.3.4\",\"port\":123,\"selfSignedCert\":\"cert\",\"options\":{\"host_sets\":\"host_set1,host_set2\"}}"
+"{\"auth\":{\"username\":\"xxx\",\"password\":  \"yyyy\"}}"
+status  {search id}
+
+#### STIX Transmit Results
+
+transmit
+trellix_endpoint_security_hx
+"{\"host\":\"1.2.3.4\",\"port\":123,\"selfSignedCert\":\"cert\",\"options\":{\"host_sets\":\"host_set1,host_set2\"}}"
+"{\"auth\":{\"username\":\"xxx\",\"password\":  \"yyyy\"}}"
+results  {search id}
+
+#### STIX Translate Results
+
+translate
+trellix_endpoint_security_hx
+results
+"{\"type\": \"identity\", \"id\": \"identity--f431f809-377b-45e0-aa1c-6a4751cae5ff\", 
+\"name\": \"trellix_endpoint_security_hx\", \"identity_class\": \"events\", \"created\": \"2023-05-30T00:00:50.336Z\", 
+\"modified\": \"2024-05-30T00:01:50.336Z\"}"
+"{transmit result response}" "{\"host_sets\": \"host_set1,host_set2\"}"
+
+
+### Recommendations
+
+- Use multiple host sets in the input to fetch records from more than 1000 hosts.
+- Progress threshold input parameter depends on the number of hosts in the host set that responds to the server. 
+  The default value of this parameter in the connector is set to 50. This value can be updated based on the number 
+  of hosts that responds.
+
+### Limitations
+
+- A maximum of 25 conditions using AND operator can be specified. Exception will be raised if more than 25 conditions 
+  are specified
```

Comment:
Is there an error message the user will receive for this scenario?

Reply:
the user will receive below message for this scenario
{
    "success": false,
    "connector": "trellix_endpoint_security_hx",
    "error": "trellix_endpoint_security_hx connector error => The maximum number of search terms is 25",
    "code": "invalid_query"
}